### ISSUE TITLE
exception during opening MF4 file

### ISSUE THREAD

**Issue by efendi54** [danielhrisca/asammdf#459]
# Python version
('python=3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:57:15) [MSC v.1915 64 bit '
 '(AMD64)]')
'os=Windows-10-10.0.17763-SP0'
'numpy=1.16.2'
'asammdf=6.1.2'

# Code 

  ## MDF version
  4.10

  ## Code snippet
  from asammdf import MDF
  with MDF(mdf_file) as mdf:
     pass
  
  ## Traceback
  File "C:\dev\AWV_TOOLS_NEW\05_Compiler_Interpreter\Python37\lib\site-packages\asammdf\mdf.py", line 147, in __init__
    self._mdf = MDF4(name, channels=channels, **kwargs)
  File "C:\dev\AWV_TOOLS_NEW\05_Compiler_Interpreter\Python37\lib\site-packages\asammdf\blocks\mdf_v4.py", line 396, in __init__
    self._read(mapped=True)
  File "C:\dev\AWV_TOOLS_NEW\05_Compiler_Interpreter\Python37\lib\site-packages\asammdf\blocks\mdf_v4.py", line 511, in _read
    history_block = FileHistory(address=fh_addr, stream=stream, mapped=mapped)
  File "C:\dev\AWV_TOOLS_NEW\05_Compiler_Interpreter\Python37\lib\site-packages\asammdf\blocks\v4_blocks.py", line 4905, in __init__
    raise MdfException(message)
asammdf.blocks.utils.MdfException: Expected "##FH" block @0x10926dd38 but found "b'\x00\x00\x00\x00'"
  
# Description
The MF4 was delivered by a third-party company and is slightly bigger than 4GB (reason why I cannot send it via email).
My application crashes with the exception above as soon as the file is tried to be opened by the code pasted above. I want to know if the file is corrupted or some other thing does not match (reason for the crashing and how if possible to fix it). 
Thnx alot.

**Comment by danielhrisca** [comment]
Most certainly the file is corrupted, at least the link to that file history block points to garbage data. 
You can check this with the freeware tool [MDFValidator from Vector Gmbh](https://www.vector.com/int/en/download/?tx_vectorproducts_productdownloaddetail%5Bdownload%5D=18928&tx_vectorproducts_productdownloaddetail%5Baction%5D=show&tx_vectorproducts_productdownloaddetail%5Bcontroller%5D=Productdownload&cHash=85e77730e9e15ce3b1e61a7e98a55485)

**Comment by efendi54** [comment]
Thnx alot for the quick reply. 

Just tried it via MDFValidator which has the problem of getting stuck itself during the file-opening-process. I tried this on two different machines with the same behavior.
I also tried another MF4 which is not corrupted (or at least does not cause any problems on my side so far) and it could be opened by the MDFValidator (without the validator getting stuck). 

So I interpret this as the initial file I had problems with as being invalid (as u had already written). 
Thnx again.
